tests/generic/test_inputlist.py

```diff
@@ -297,79 +301,162 @@ def rec(m):
+    def test_to_hdf_items(self):
+        """
+        Should write all sublists to HDF groups and simple items to HDF
+        datasets.
+        """
+        self.pl.to_hdf(hdf=self.hdf)
+        for i, (k, v) in enumerate(self.pl.items()):
+            k = "{}__index_{}".format(k if isinstance(k, str) else "", i)
+            if isinstance(v, InputList):
+                self.assertTrue(k in self.hdf["input"].list_groups(),
+                                "Sublist '{}' not a sub group in hdf!".format(k))
+            else:
+                self.assertTrue(k in self.hdf["input"].list_nodes(),
+                                "Item '{}' not a dataset in hdf!".format(k))
+
+    def test_to_hdf_name(self):
+        """
+        Should raise error if clashing names are given.
+        """
+        with self.assertRaises(ValueError,
+                msg="Cannot have names clashing with index mangling."):
+            InputList({'__index_0': 42}).to_hdf(hdf=self.hdf)
 
     def test_to_hdf_group(self):
+        """
+        Should be possible to give a custom group name.
+        """
         self.pl.to_hdf(hdf=self.hdf, group_name = "test_group")
         self.assertEqual(self.hdf["test_group/NAME"],
                          "InputList")
         self.assertEqual(self.hdf["test_group/TYPE"],
                          "<class 'pyiron_base.generic.inputlist.InputList'>")
         self.assertEqual(self.hdf["test_group/OBJECT"],
                          "InputList")
-        l = InputList(self.hdf["test_group/data"])
-        self.assertEqual(self.pl, l)
 
     def test_to_hdf_readonly(self):
+        """
+        Read-only property should be stored.
+        """
         self.pl.to_hdf(hdf=self.hdf, group_name = "read_only_f")
-        self.assertTrue("read_only" in self.hdf.list_nodes(),
+        self.assertTrue("READ_ONLY" in self.hdf["read_only_f"].list_nodes(),
                         "read-only parameter not saved in HDF")
         self.assertEqual(self.pl.read_only,
-                         self.hdf[self.pl.table_name]["read_only"],
+                         self.hdf[self.pl.table_name]["READ_ONLY"],
                          "read-only parameter not correctly written to HDF")
 
         pl = self.pl.copy()
         pl.read_only = True
         pl.to_hdf(hdf=self.hdf, group_name = "read_only_t")
         self.assertEqual(pl.read_only,
-                         self.hdf["read_only_t/read_only"],
+                         self.hdf["read_only_t/READ_ONLY"],
                          "read-only parameter not correctly written to HDF")
 
     def test_from_hdf(self):
+        """
+        Reading from HDF should give back the same list as written.
+        """
         self.pl.to_hdf(hdf=self.hdf)
         l = InputList(table_name = "input")
         l.from_hdf(hdf=self.hdf)
         self.assertEqual(self.pl, l)
 
     def test_from_hdf_group(self):
+        """
+        Reading from HDF should give back the same list as written even with
```

Comment:
Nit: @samwaseda and I concluded over in [PR 1163](https://github.com/pyiron/pyiron/pull/1163#event-3982966405) that we would intentionally violate PEP8's 79 character line limit in favour of the pycharm default of 120 characters.